xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -612,3 +674,22 @@ bool CGUIWindowPVRGuide::OnContextButtonDeleteTimer(CFileItem *item, CONTEXT_BUT
 
   return bReturn;
 }
+
+CPVRRefreshTimelineItemsThread::CPVRRefreshTimelineItemsThread(CGUIWindowPVRGuide *pGuideWindow)
+: CThread("epg-grid-refresh-timeline-items"),
+  m_pGuideWindow(pGuideWindow)
+{
+}
+
+void CPVRRefreshTimelineItemsThread::Process()
+{
+  while (!m_bStop)
+  {
+    if (m_pGuideWindow->RefreshTimelineItems() && !m_bStop)
```

Comment:
It's there by intention, because RefreshTimeLineItems may take long time I thought it's worth checking afterwards whether it still makes sense to trigger another refresh of the window.